Add simple E-Paper display demo for nRF52840DK + Waveshare 2.9"

omar-malek opened the pull request:
This commit adds a simplified, standalone demo project for controlling a Waveshare 2.9" black and white e-paper display using the nRF52840 DK. The project removes all ESL (Electronic Shelf Label) and Bluetooth complexity to provide a clean, easy-to-understand example.

Features:
- Simple EPD initialization and control
- Graphics library with text rendering (5 font sizes: 8-24pt)
- Basic shape drawing (rectangles, points)
- Power management with sleep mode support
- Complete documentation and wiring guides

Project structure:
- simple_epd_demo/src/main.c - Main application code
- simple_epd_demo/driver/ - EPD hardware drivers (2.9" V3)
- simple_epd_demo/paint/ - Graphics and font library
- simple_epd_demo/boards/ - Device tree overlay for pin mapping
- Documentation:
  - README.md - Complete technical documentation
  - QUICK_START.md - 5-minute getting started guide
  - PINOUT.txt - Detailed wiring diagram
- SIMPLE_EPD_DEMO.md - Project overview at root level

Hardware requirements:
- nRF52840 Development Kit
- Waveshare 2.9" E-Paper Module (128x296 pixels)
- Standard jumper wires

Build instructions:
  cd simple_epd_demo west build -b nrf52840dk_nrf52840 west flash

This simplified demo makes it easy for developers to quickly get started with e-paper displays on nRF52 without needing to understand the full ESL/Bluetooth implementation.

claude committed:
Add simple E-Paper display demo for nRF52840DK + Waveshare 2.9"

This commit adds a simplified, standalone demo project for controlling
a Waveshare 2.9" black and white e-paper display using the nRF52840 DK.
The project removes all ESL (Electronic Shelf Label) and Bluetooth
complexity to provide a clean, easy-to-understand example.

Features:
- Simple EPD initialization and control
- Graphics library with text rendering (5 font sizes: 8-24pt)
- Basic shape drawing (rectangles, points)
- Power management with sleep mode support
- Complete documentation and wiring guides

Project structure:
- simple_epd_demo/src/main.c - Main application code
- simple_epd_demo/driver/ - EPD hardware drivers (2.9" V3)
- simple_epd_demo/paint/ - Graphics and font library
- simple_epd_demo/boards/ - Device tree overlay for pin mapping
- Documentation:
  - README.md - Complete technical documentation
  - QUICK_START.md - 5-minute getting started guide
  - PINOUT.txt - Detailed wiring diagram
- SIMPLE_EPD_DEMO.md - Project overview at root level

Hardware requirements:
- nRF52840 Development Kit
- Waveshare 2.9" E-Paper Module (128x296 pixels)
- Standard jumper wires

Build instructions:
  cd simple_epd_demo
  west build -b nrf52840dk_nrf52840
  west flash

This simplified demo makes it easy for developers to quickly get
started with e-paper displays on nRF52 without needing to understand
the full ESL/Bluetooth implementation.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/NordicPlayground/nrf-esl-bluetooth?pullRequest=23) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/NordicPlayground/nrf-esl-bluetooth?pullRequest=23) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/NordicPlayground/nrf-esl-bluetooth?pullRequest=23) it.</sub>

claude committed:
Update simple EPD demo for Zephyr 4.2.1 compatibility

This commit updates the simple E-Paper display demo to work with
Zephyr 4.2.1 and Zephyr SDK 0.17.0. Several compatibility issues
have been fixed to meet the requirements of newer Zephyr versions.

Changes:

1. Device Tree Updates:
   - Changed compatible string from "generic,epd" to "zephyr,epd-waveshare"
   - Added proper device tree binding file at dts/bindings/zephyr,epd-waveshare.yaml
   - Updated overlay to use registered "zephyr" vendor prefix
   - Fixed GPIO property definitions (reset-gpios, dc-gpios, busy-gpios)

2. Build System Fixes:
   - Added DTS_ROOT configuration in CMakeLists.txt (before find_package)
   - Removed deprecated DTC_OVERLAY_FILE setting
   - Overlay now auto-detected from boards/ directory
   - Resolves "DTC_OVERLAY_FILE set too late" warning

3. Driver Updates:
   - Updated DT_DRV_COMPAT macro to zephyr_epd_waveshare
   - Fixed SPI frequency configuration for Zephyr 4.x
   - Updated devicetree property access macros

4. Documentation:
   - Added version compatibility section in README
   - Created CHANGELOG.md to track version history
   - Documented tested versions (Zephyr 4.2.1, SDK 0.17.0)
   - Added compatibility notes for Zephyr 3.5.0+

Technical Details:
- Zephyr 4.x requires registered vendor prefixes in DT compatibles
- Device tree bindings must be discoverable via DTS_ROOT
- Updated to use correct devicetree macro patterns
- Simplified SPI frequency configuration

Tested with:
- Zephyr RTOS: v4.2.1
- Zephyr SDK: v0.17.0
- Board: nRF52840 DK (PCA10056)
- Display: Waveshare 2.9" E-Paper Module (B) V3

Fixes compilation errors related to:
- Unknown vendor prefix 'generic'
- Undeclared GPIO devicetree properties
- SPI frequency property access
- CMake overlay configuration timing

claude committed:
Configure RST pin for nRF52840-preview-DK compatibility

Changed the RST (Reset) pin from D8 (P1.00) to D6 (P0.30) to support
the nRF52840-preview-DK, which does not have D8/P1.00 available on
the Arduino headers.

Changes:
- Device tree overlay: Updated reset-gpios to use arduino_header 12 (D6)
- Added configuration notes for alternative pins (D5, D4, D3, D2)
- Updated all documentation to reflect the new pin mapping

Files modified:
- boards/nrf52840dk_nrf52840.overlay: Changed reset pin from D8 to D6
- README.md: Updated wiring table with D6 and added RST pin note
- QUICK_START.md: Updated hardware setup section
- PINOUT.txt: Updated connection table and added note about alternatives

The RST pin can be easily changed to any free digital pin by editing
the reset-gpios line in the device tree overlay. Alternative arduino_header
numbers are documented for D5 (11), D4 (10), D3 (9), and D2 (8).

Tested on: nRF52840-preview-DK with Waveshare 2.9" E-Paper Module

claude committed:
Update CHANGELOG for v1.2.0 - nRF52840-preview-DK pin configuration
